BurdaMagazinOrg/module-fb_instant_articles
# Issue #2872807 Manage creation and rendering of InstantArticle object via the Serialization API

This PR introduces Normalizers and an Encoder to integrated the FBIA PHP SDK with the Drupal Serialzation API. It allows you to programatically load a node, pipe it through the serialization API, and get FBIA markup out the other end. As an argument you pass an EntityViewDisplay object as a mapping to guide the process.

In order to manually test:

* Enable `fb_instant_articles` module
* Enable Facebook Instant Articles support on one of your content types (eg. `/admin/structure/types/manage/article`).
* Configure the `fb_instant_articles` view mode (eg. `/admin/structure/types/manage/article/display/fb_instant_articles`) by putting fields into regions and assigning FBIA prefixed field formatters.
* Open up a PHP interpreter using `drush php` and enter the following:

```
>>> \Drupal::service('serializer')->serialize(\Drupal\node\Entity\Node::load(1), 'fbia', ['entity_view_display' => \Drupal::entityTypeManager()->getStorage('entity_view_display')->load('node.article.fb_instant_articles')]);
```
This invokes the `serialize` method on the `serializer` service, passing in a node object (nid 1), the format (`fbia`) and the EntityViewDisplay object. This may differ slightly for your own local setup.

Also run the test suite to verify that everything is green, eg.

```
$ cd core
$ ../vendor/bin/phpunit --verbose --debug --group=fb_instant_articles
```
